parcels/particlefile.py

```diff
@@ -31,43 +27,30 @@ class ParticleFile(object):
     :param outputdt: Interval which dictates the update frequency of file output
                      while ParticleFile is given as an argument of ParticleSet.execute()
                      It is either a timedelta object or a positive double.
-    :param type: Either 'array' for default matrix style, or 'indexed' for indexed ragged array
     :param write_ondelete: Boolean to write particle data only when they are deleted. Default is False
     """
 
-    def __init__(self, name, particleset, outputdt=np.infty, type='array', write_ondelete=False):
+    def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False):
 
-        self.type = type
         self.name = name
         self.write_ondelete = write_ondelete
         self.outputdt = outputdt
-        if self.write_ondelete and self.type is 'array':
-            logger.warning('ParticleFile.write_ondelete=True requires type="indexed". Setting that option')
-            self.type = 'indexed'
         self.outputdt = outputdt
         self.lasttime_written = None  # variable to check if time has been written already
-        self.dataset = netCDF4.Dataset("%s.nc" % name, "w", format="NETCDF4")
+        extension = path.splitext(str(name))[1]
+        fname = name if extension in ['.nc', '.nc4'] else "%s.nc" % name
+        self.dataset = netCDF4.Dataset(fname, "w", format="NETCDF4")
         self.dataset.createDimension("obs", None)
-        if self.type is 'array':
-            self.dataset.createDimension("trajectory", particleset.size)
-            coords = ("trajectory", "obs")
-        elif self.type is 'indexed':
-            coords = ("obs")
-        else:
-            raise RuntimeError("ParticleFile type must be either 'array' or 'indexed'")
+        self.dataset.createDimension("trajectory", None)
```

Comment:
It is not a comment concerning the changes, but why do we call the particle dimension "trajectory", and not "particle" or "id"?

Reply:
That's because we want to stay CF-compliant: https://www.nodc.noaa.gov/data/formats/netcdf/v2.0/trajectoryIncomplete.cdl